### DISCUSSION TITLE
Eaton/Aspire RFWC5 5-button Controller Scene Deactivate Event

### DISCUSSION THREAD

**Discussion by apetrycki** [zwave-js/zwave-js#3391] [Q&A]
I'm trying to find a way to get Home Assistant to recognize that a button press has occurred to deactivate a scene.  When a scene is deactivated, I get:
```
16:51:41.143 CNTRLR [Node 019] treating BasicCC::Set as a report
2021-09-22 16:51:41.146 INFO ZWAVE: Node 19: value updated: 32-0-currentValue 0 => 0
```
HA does not see this as an event or update.  "Basic" is not seen in the network dump.  What I would like to do is trigger an Indicator LED refresh when this event occurs like I used to be able to do with the old Z-Wave integration, which would then trigger an automation to turn off the scene.  

Why are the Basic CC target/current value IDs missing, or should the device treat Basic Set as an event?

[zwave_js_dump.json.txt](https://github.com/zwave-js/node-zwave-js/files/7212767/zwave_js_dump.json.txt)

**Comment by AlCalzone** [comment]
From your dump it looks like the device uses `Scene Activation CC` to communicate which button was pressed. Don't you get any event when the scene should be deactivated?

A more complete [driver log](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/bug_report?id=driver-logs) than just these two lines would be helpful.

**Reply by apetrycki** [reply]
Here's my log.  I enabled MQTT and now use that through Node-Red to catch the deactivation and do a refresh to get the indicator value.  The indicator sensor value in HA also shows up as unavailable until I do a refresh from within Z-Wave JS.  The log shows me pushing button 1 which triggers scene 1.  I catch that through MQTT and refresh the indicator sensor which triggers another automation that grabs the indicator value and determines which light to turn on.  It turns on a zigbee light.  I then push the button again, which triggers the above message and grab that through MQTT, do another indicator refresh, and the same automation to determine which light to turn off.  It then turns the zigbee light back off.

Let me know if you need anything else.

[zwave_js.log](https://github.com/zwave-js/node-zwave-js/files/7217464/zwave_js.log)

**Reply by AlCalzone** [reply]
I don't get the point of always querying the indicator. You receive two value events for `sceneId` which tells you which button was pressed:
```
2021-09-23T12:19:01.444Z CNTRLR   [Node 019] [!] [Scene Activation] sceneId: 1                      [Endpoint 0]
2021-09-23T12:19:01.466Z CNTRLR   [Node 019] [!] [Scene Activation] sceneId: 1                      [Endpoint 0]
```

I don't see anything for the deactivation though, but maybe that's just a configuration issue (I don't have that device and I don't seem to be able to find a proper manual). Since 8.3.0, you should be able to properly configure Scene Controller Configuration CC which your device seems to support, and tell it which button should send which scene ID.

If that doesn't help with turning off, we can always enable a compat flag to prevent the Basic CC from being hidden.

**Reply by apetrycki** [reply]
Yeah, I can grab the scene on activation, but then I'd need 2 automations instead of just the 1.  I just look at the indicator, compare it to previous, convert it to binary, and use the 1 or 0 to turn on/off the device.  It makes the code pretty easy.

As far as I can tell, everything is configured correctly.  It reports the scene when I activate, just nothing on deactivation, which is why I have to use the indicator lights.  Other people seem to have the same issue.  Exposing the Basic CC would certainly help.  At least then people can use HA to do what I'm doing over MQTT.  Do you have sample output from a controller that does report which scene was deactivated?  Any other devices similar to this Eaton one that maybe works better?  I'm looking to get some more for other rooms, so if there are other wired scene controllers that are more compatible, I'd rather go with those.

I also noticed that Z-Wave JS advertises sensors for the scene and scene delay but they don't show up in HA.  Is this just something that's not supported yet?  I tried doing a rediscover.

**Reply by apetrycki** [reply]
Also, any idea why the messages are received twice?  I always get 2 reports of the same value over MQTT.

**Reply by AlCalzone** [reply]
> but then I'd need 2 automations instead of just the 1

True, but you send less commands over a very constrained network.

> Do you have sample output from a controller that does report which scene was deactivated?

Nope, sorry. I only have a handful of devices and scene controllers aren't very popular.

> I also noticed that Z-Wave JS advertises sensors for the scene and scene delay but they don't show up in HA. Is this just something that's not supported yet? I tried doing a rediscover.

You'll have to ask that in the HA repo. The driver isn't concerned with HA's sensors.

> Also, any idea why the messages are received twice? I always get 2 reports of the same value over MQTT.

You probably have the controller twice in the node's association groups (or 255, the broadcast node).

**Reply by kpine** [reply]
> > I also noticed that Z-Wave JS advertises sensors for the scene and scene delay but they don't show up in HA. Is this just something that's not supported yet? I tried doing a rediscover.
> 
> You'll have to ask that in the HA repo. The driver isn't concerned with HA's sensors.

For one reason, Scene Activation notifications are usually handled as events in HA, so there's no need for a sensor. But I think the other reason is that the `sceneId` value ID is value-less. The driver provides the value ID and metadata, but the value ID has no `value`, at least that's the case from the provided dump. Without a `value`, there's no state for a sensor.

**Reply by AlCalzone** [reply]
> But I think the other reason is that the sceneId value ID is value-less.

That's also because it is handled as an event.

**Reply by apetrycki** [reply]
Thanks for the PR and all the info!

**Comment by AlCalzone** [comment]
https://github.com/zwave-js/node-zwave-js/issues/3398 and corresponding PR https://github.com/zwave-js/node-zwave-js/pull/3399
